src/Symfony/Component/Messenger/Transport/Serialization/RemoveNonSendableStampsTrait.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Serialization;
+
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Stamp\NonSendableStampInterface;
+
+trait RemoveNonSendableStampsTrait
```

Comment:
it's missing your author and experimental annotation

Reply:
what about replacing this trait with a new method on Envelope?

Reply:
Could be `withoutStampsOfType(string $fqcn);`